test/test_pipeline_yaml.py

```diff
@@ -792,3 +847,45 @@ def test_save_yaml_overwrite(tmp_path):
     with open(tmp_path / "saved_pipeline.yml", "r") as saved_yaml:
         content = saved_yaml.read()
         assert content != ""
+
+
+def test_save_yaml_with_custom_node_kwargs(tmp_path):
+    class CustomNode(MockNode):
+        def __init__(self, param: int):
+            super().__init__()
+            self.param = param
+
+    pipeline = Pipeline()
+    pipeline.add_node(CustomNode(param=1), name="custom_node", inputs=["Query"])
+    pipeline.save_to_yaml(tmp_path / "saved_pipeline.yml")
+
+    print(pipeline.get_node("custom_node")._component_config)
+
+    with open(tmp_path / "saved_pipeline.yml", "r") as saved_yaml:
+        content = saved_yaml.read()
+
+        assert content.count("custom_node") == 2
+        assert "CustomNode" in content
+        assert "param: 1" in content
+        assert "Query" in content
+        assert f"version: {haystack.__version__}" in content
+
+
+def test_save_yaml_with_custom_node_args(tmp_path):
+    class CustomNode(MockNode):
+        def __init__(self, param: int):
+            super().__init__()
+            self.param = param
+
+    pipeline = Pipeline()
+    pipeline.add_node(CustomNode(1), name="custom_node", inputs=["Query"])
+    pipeline.save_to_yaml(tmp_path / "saved_pipeline.yml")
+
+    with open(tmp_path / "saved_pipeline.yml", "r") as saved_yaml:
+        content = saved_yaml.read()
+
+        assert content.count("custom_node") == 2
+        assert "CustomNode" in content
+        assert not "param: 1" in content
```

Comment:
At first I thought this is a mistake, but I then got that we currently do not support unnamed positional args. I was wondering whether we want a test like this that tests for a "feature not being implemented" at all. At least I think we should comment on that, just in case we want to support unnamed positional args in the future and this test starts to fail. Otherwise I can imagine the one implementing the feature to be pretty puzzled about this test.

Reply:
Good point, I didn't think of that. Now I'm also not sure anymore whether this test is useful or not... I think the right thing to do here would be to ensure that the warning is printed, but it's too complex for too little of a benefit. I think I'll just remove this :+1:

Reply:
It's actually not very complex to check for logs: @bogdankostic did this in https://github.com/deepset-ai/haystack/pull/2369/files using the `caplog` fixture

Reply:
Looking into it!

Reply:
Done :+1: looking much better now indeed